docs/getting-started/workflows.md

```diff
@@ -0,0 +1,340 @@
+
+1. In code — by defining them as workflow functions and registering them via `LazyWorkflowInstance`.
+2. In the database — by mapping them to the corresponding `product_type` using a migration.
+    - workflows don't need to necessarily be added to a product_type, doing this will only make them available as tasks not meant to be ran by a subscription.
+
+We’ll start with the code registration, followed by options for generating the database migration.
+
+### Step 1: Register workflow functions in code
+
+Workflow functions must be registered by creating a `LazyWorkflowInstance`, which maps a workflow function to the Python module where it's defined.
+
+Example — registering the `create_user_group` workflow:
+
+```python
+from orchestrator.workflows import LazyWorkflowInstance
+
+LazyWorkflowInstance("workflows.user_group.create_user_group", "create_user_group")
+```
+
+To ensure the workflows are discovered at runtime:
+
+- Add all `LazyWorkflowInstance(...)` calls to `workflows/__init__.py`.
+- Add `import workflows` to `main.py` so they are registered during app startup.
+
+!!! example
+
+    for inspiration look at an example implementation of the [lazy workflow instances]
+
+### Step 2: Register workflows in the database
+
+After registering workflows in code, you need to add them to the database by mapping them to their `product_type`.
+There are three ways to do this:
+
+- [Migrate workflows generator script](#migrate-workflows-generator-script)
+- [Copy the example workflows migration](#copy-the-example-workflows-migration)
+- [Manual](#manual)
+
+#### Migrate workflows generator script
+
+Similar to `db migrate-domain-models`, the orchestrator command line interface offers the `db migrate-workflows` command
+that walks you through a menu to create a database migration file based on the difference between the registered workflows in the code and the database.
+
+Start with the following command:
+
+```shell
+python main.py db migrate-workflows "add User and UserGroup workflows"
+```
+
+Navigate through the menu to add the six workflows to the corresponding `User` or `UserGroup` product type.
+After confirming a migration file will be added to `migrations/versions/schema`.
+
+The migration can be run with:
+
+```shell
+python main.py db upgrade heads
+```
+
+#### Copy the example workflows migration
+
+You can copy a predefined migration file from the example repository:
+
+```shell
+(
+  cd migrations/versions/schema
+  curl --remote-name https://raw.githubusercontent.com/workfloworchestrator/example-orchestrator-beginner/main/examples/2022-11-12_8040c515d356_add_user_and_usergroup_workflows.py
+)
+```
+
+Update it to your own workflow and update the database with:
+
+```shell
+python main.py db upgrade heads
+```
+
+#### Manual
+
+Create a new empty database migration with the following command:
+
+```shell
+PYTHONPATH=. python main.py db revision --head data --message "add User and UserGroup workflows"
```

Comment:
Think `PYTHONPATH=.` is a remnant from the past? I haven't had to use this in our own nor the example orchestrator. Imo it'd be better to have this in a troubleshooting section (if people indeed run into it)

```suggestion
python main.py db revision --head data --message "add User and UserGroup workflows"
```